app/com/arpnetworking/metrics/portal/scheduling/JobExecutorActor.java

```diff
@@ -96,42 +97,65 @@ public void preStart() throws Exception {
      * @param wakeUpBy The time we want to be sure we wake up by.
      * @return The time until we should wake up. Always between 0 and (approximately) {@code wakeUpBy}.
      */
-    /* package private */ FiniteDuration timeUntilNextTick(final Instant wakeUpBy) {
+    /* package private */ Optional<FiniteDuration> timeUntilExtraTick(final Instant wakeUpBy) {
         final FiniteDuration delta = Duration.fromNanos(ChronoUnit.NANOS.between(_clock.instant(), wakeUpBy));
         if (delta.lt(Duration.Zero())) {
-            return Duration.Zero();
+            return Optional.of(Duration.Zero());
         } else if (delta.lt(TICK_INTERVAL)) {
-            return delta;
-        } else {
-            return TICK_INTERVAL;
+            return Optional.of(delta);
         }
+        return Optional.empty();
     }
 
-    private void scheduleNextTick(final Instant wakeUpBy) {
-        timers().startSingleTimer("TICK", Tick.INSTANCE, timeUntilNextTick(wakeUpBy));
+    private void scheduleExtraTickIfNecessary(final Instant wakeUpBy) {
+        timeUntilExtraTick(wakeUpBy).ifPresent(timeRemaining -> {
+            timers().startSingleTimer("TICK", Tick.INSTANCE, timeRemaining);
+        });
     }
 
-    private void executeAndScheduleNextTick(
-            final JobRepository<T> repo,
-            final Organization org,
-            final Job<T> job,
-            final Instant scheduled
-    ) {
-        final Optional<Instant> nextScheduled = job.getSchedule().nextRun(Optional.of(scheduled));
-        repo.jobStarted(job.getId(), org, scheduled);
-        job.execute(getSelf(), scheduled)
-                .handle((result, error) -> {
-                    if (error == null) {
-                        repo.jobSucceeded(job.getId(), org, scheduled, result);
-                    } else {
-                        repo.jobFailed(job.getId(), org, scheduled, error);
-                    }
-                    return null;
-                })
-                .thenApply(whatever -> {
-                    nextScheduled.ifPresent(this::scheduleNextTick);
-                    return null;
-                });
+    private void stopForever() {
```

Comment:
This isn't entirely true right? I mean it's just until the coordinator resurrects it?